source/auth/auth.rst

```diff
@@ -1252,7 +1252,10 @@ mechanism
 mechanism_properties
     PROVIDER_NAME
         Drivers MUST allow the user to specify a name for using a service
-        to obtain credentials that is one of ["aws"].
+        to obtain credentials that is one of ["aws", "azure"].
+    TOKEN_AUDIENCE
```

Comment:
Question here that would negate my other questions. Will the resource we pass always be `api://${AZUREOIDC_CLIENTID}`? I'm curious if this mech property can be removed and we just read `AZUREOIDC_CLIENTID` from the environment and construct the resource parameter we send to the Azure endpoint.

Reply:
I wrestled with whether this should be an env variable or a mech property.  The GCP integration needs a similar value.   The user would have to set the environment variable on their application, so we'd want to choose something that wouldn't conflict with other values, perhaps `MONGODB_OIDC_TOKEN_AUDIENCE`?

![image](https://github.com/mongodb/specifications/assets/2096628/2fbb36fa-ab17-49fd-ad0b-0a1106f1f667)

Reply:
Ah I guess if the user needs to manually set it then I'm less worried about it being in the URI vs an environment variable. So I think URI is fine then.